### PR TITLE
Better document a constructor of FE_Q.

### PR DESCRIPTION
**Pull request by bangerth**
Had to look this up for a side-conversation about finite elements. Thought I'd document why this constructor exists to begin with.

### CODE REVIEW

File: include/deal.II/fe/fe_q.h [hunk 1]
```diff
@@ -564,6 +564,36 @@ class FE_Q : public FE_Q_Base<TensorProductPolynomials<dim>, dim, spacedim>
    * constructor that specifies the polynomial degree only. For selecting
    * equidistant nodes at <tt>fe_degree > 2</tt>, construct
    * <tt>FE_Q<dim>(QIterated<1>(QTrapezoid<1>(),fe_degree))</tt>.
+   *
+   * This constructor results in an FE_Q object whose basis functions are still
+   * polynomials of degree equal to the number of elements of the @p points
```
Comment (tjhei) [inline include/deal.II/fe/fe_q.h:569]: This sentence seems to be redundant to me, as it says the same thing as "The degree of the ..." above.

File: include/deal.II/fe/fe_q.h [hunk 2]
```diff
@@ -564,6 +564,36 @@ class FE_Q : public FE_Q_Base<TensorProductPolynomials<dim>, dim, spacedim>
...
+   * argument minus one. In other words, the *space* so constructed is the
+   * same as if you had called `FE_Q<dim>(points.size()-1)`. What *is*
```
Comment (tjhei) [inline include/deal.II/fe/fe_q.h:571]: Are you sure ``*is*`` renders correctly in doxygen?

**Review by tjhei — COMMENTED** [review]
Very nice explanation!

**Review by masterleinad — APPROVED** [review]

File: include/deal.II/fe/fe_q.h (hunk 2 above)
Comment (bangerth) [reply include/deal.II/fe/fe_q.h:571]: Pretty sure -- doxygen does markdown these days. We've used in other places as well in recent times.

**Comment by bangerth** [comment]
@tjhei Fixed as suggested.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**